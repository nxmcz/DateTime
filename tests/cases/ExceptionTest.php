<?php

declare(strict_types=1);

namespace Tests\Unit;

use Noxem\DateTime\Exception\BadFormatException;
use Tester\Assert;
use Tests\Fixtures\TestCase\AbstractTestCase;

require_once __DIR__ . '/../bootstrap.php';


/**
 * @testCase
 */
class ExceptionTest extends AbstractTestCase
{
	public function testBadFormatException(): void
	{
		Assert::exception(
			fn() => BadFormatException::create(),
			BadFormatException::class,
			"Bad DateTimeImmutable constructors format. Must be INT >= 32400 or string-date in format like: YYYY-MM-DD H:i:s"
		);
	}
}

$test = new ExceptionTest();
$test->run();