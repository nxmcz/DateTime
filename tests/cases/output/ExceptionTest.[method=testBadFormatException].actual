Noxem\DateTime\Exception\BadFormatException::__set_state(/* #c4b4 */ [
	'message' => 'Bad DateTimeImmutable constructors format. Must be INT >= 32400 or string-date in format like: YYYY-MM-DD H:i:s',
	'string' => '',
	'code' => 0,
	'file' => 'C:\repositories\date-time\src\Exception\BadFormatException.php',
	'line' => 10,
	'trace' => [
		[
			'file' => 'C:\repositories\date-time\tests\cases\ExceptionTest.php',
			'line' => 23,
			'function' => 'create',
			'class' => 'Noxem\DateTime\Exception\BadFormatException',
			'type' => '::',
			'args' => [],
		],
		[
			'file' => 'C:\repositories\date-time\vendor\nette\tester\src\Framework\TestCase.php',
			'line' => 126,
			'function' => 'testBadFormatException',
			'class' => 'Tests\Unit\ExceptionTest',
			'type' => '->',
			'args' => [],
		],
		[
			'file' => 'C:\repositories\date-time\vendor\nette\tester\src\Framework\TestCase.php',
			'line' => 51,
			'function' => 'runTest',
			'class' => 'Tester\TestCase',
			'type' => '->',
			'args' => [
				ReflectionMethod::__set_state(/* #7dbd */ [
					'name' => 'testBadFormatException',
					'class' => 'Tests\Unit\ExceptionTest',
				]),
			],
		],
		[
			'file' => 'C:\repositories\date-time\tests\cases\ExceptionTest.php',
			'line' => 29,
			'function' => 'run',
			'class' => 'Tester\TestCase',
			'type' => '->',
			'args' => [],
		],
	],
	'previous' => null,
])
