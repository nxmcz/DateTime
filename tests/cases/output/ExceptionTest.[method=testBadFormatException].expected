Bad DateTimeImmutable constructors format. Must be INT >= 32400 or string-date in format like: YYYY-MM-DD H:i:s